<?xml version="1.0"?>

<launch>
  <arg name="gui" default="true"/>
  <arg name="world_name" default="test_easy"/>
  <arg name="robot_num" default="3"/>
  <arg name="stage_index" default="3"/>

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find detour)/worlds/$(arg world_name).world"/>
    <arg name="gui" value="$(arg gui)"/>
  </include>

  <include file="$(find husky_gazebo)/launch/spawn_husky.launch">
    <arg name="robot_namespace" value="robot0"/>
    <arg name="x" value="15" />
    <arg name="y" value="3" />
    <arg name="z" value="0.2" />
    <arg name="board_color" value="Red" />
  </include>

  <include file="$(find husky_gazebo)/launch/spawn_husky.launch">
    <arg name="robot_namespace" value="robot1"/>
    <arg name="x" value="15" />
    <arg name="y" value="0" />
    <arg name="z" value="0.2" />
    <arg name="board_color" value="Green" />
  </include>

  <include file="$(find husky_gazebo)/launch/spawn_husky.launch">
    <arg name="robot_namespace" value="robot2"/>
    <arg name="x" value="15" />
    <arg name="y" value="-3" />
    <arg name="z" value="0.2" />
    <arg name="board_color" value="Blue" />
  </include>


  <node pkg="detour" type="tf_transfer_world.py" name="tf_transfer_world">
    <param name="robot_num" value="$(arg robot_num)"/>
    <param name="stage_index" value="$(arg stage_index)"/>
  </node>

  <node pkg="detour" type="path_pub_node.py" name="path_pub_node">
    <param name="robot_num" value="$(arg robot_num)"/>
  </node>

  <node pkg="detour" type="obstacle_pub_node.py" name="obstacle_pub_node">
    <param name="stage_index" value="$(arg stage_index)"/>
  </node>

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find detour)/rviz/config2.rviz "/>
  <node pkg="rqt_publisher" type="rqt_publisher" name="rqt_publisher" />
  <node pkg="rqt_topic" type="rqt_topic" name="rqt_topic" />

</launch>